stratosphere/pm/source/impl/pm_process_manager.cpp

```diff
@@ -0,0 +1,680 @@
+                g_process_event->Signal();
+                break;
+            case ProcessState_RunningAttached:
+                if (process_info->ShouldSignalOnDebugEvent()) {
+                    process_info->ClearSuspended();
+                    process_info->SetSuspendedStateChanged();
+                    g_process_event->Signal();
+                }
+                break;
+            case ProcessState_Exited:
+                if (GetRuntimeFirmwareVersion() < FirmwareVersion_500 && process_info->ShouldSignalOnExit()) {
+                    g_process_event->Signal();
+                } else {
+                    ProcessListAccessor list(g_process_list);
+                    CleanupProcess(list, process_info);
+                }
+                /* Return ConnectionClosed to cause libstratosphere to stop waiting on the process. */
+                return ResultKernelConnectionClosed;
+            case ProcessState_DebugSuspended:
+                if (process_info->ShouldSignalOnDebugEvent()) {
+                    process_info->SetSuspended();
+                    process_info->SetSuspendedStateChanged();
+                    g_process_event->Signal();
+                }
+                break;
+        }
+
+        return ResultSuccess;
+    }
+
+    /* Process Management. */
+    Result LaunchTitle(u64 *out_process_id, const ncm::TitleLocation &loc, u32 flags) {
+        /* Ensure we only try to launch one title at a time. */
+        static HosMutex s_lock;
+        std::scoped_lock<HosMutex> lk(s_lock);
+
+        /* Set global arguments, signal, wait. */
+        g_process_launch_args = {
+            .out_process_id = out_process_id,
+            .location = loc,
+            .flags = flags,
+        };
+        g_process_launch_finish_signal.Reset();
+        g_process_launch_start_event->Signal();
+        g_process_launch_finish_signal.Wait();
+
+        return g_process_launch_result;
+    }
+
+    Result StartProcess(u64 process_id) {
+        ProcessListAccessor list(g_process_list);
+
+        auto process_info = list->Find(process_id);
+        if (process_info == nullptr) {
+            return ResultPmProcessNotFound;
+        }
+
+        if (process_info->HasStarted()) {
+            return ResultPmAlreadyStarted;
+        }
+
+        ldr::ProgramInfo program_info;
+        R_TRY(ldr::pm::GetProgramInfo(&program_info, process_info->GetTitleLocation()));
+        return StartProcess(process_info, &program_info);
+    }
+
+    Result TerminateProcess(u64 process_id) {
+        ProcessListAccessor list(g_process_list);
+
+        auto process_info = list->Find(process_id);
+        if (process_info == nullptr) {
+            return ResultPmProcessNotFound;
+        }
+
+        return svcTerminateProcess(process_info->GetHandle());
+    }
+
+    Result TerminateTitle(ncm::TitleId title_id) {
+        ProcessListAccessor list(g_process_list);
+
+        auto process_info = list->Find(title_id);
```

Comment:
beware adding an implicit constructor to `ncm::TitleId` and messing up this overload

Reply:
ncm::TitleId must never have a constructor of any kind, because it must be pod.